src/vkscript/command_parser.cc

```diff
@@ -470,13 +470,36 @@ Result CommandParser::ProcessClear() {
 
 Result CommandParser::ParseValues(const std::string& name,
                                   const DatumType& type,
-                                  std::vector<Value>* values) {
+                                  std::vector<Value>* values,
+                                  bool use_std430_layout = false) {
   assert(values);
 
+  uint32_t row_index = 0;
   auto token = tokenizer_->NextToken();
+  size_t seen = 0;
   while (!token->IsEOL() && !token->IsEOS()) {
     Value v;
 
+    // When |type| is a multi-column matrix, fill more values for its
+    // memory layout.
+    if (type.ColumnCount() > 1U && type.RowCount() > 1U &&
+        row_index == type.RowCount()) {
+      if (use_std430_layout || type.RowCount() == 3U) {
+        if ((type.IsFloat() || type.IsDouble())) {
+          v.SetDoubleValue(0);
+        } else {
+          v.SetIntValue(0);
+        }
+        // For std430 layout, the stride of each row must be
+        // 4 * element size in bytes when row count is not 1.
+        // For std140 layout, the stride of each row must be
+        // 4 * element size in bytes when row count is 3 or 4.
```

Comment:
In std140, if row size is 2 can the stride be 16? Can we just always make the row stride 16 and not worry if it's std430 or 140?

Reply:
No.
If it is std140 and the row size is 2, its stride will be 8.
Please check this condition `use_std430_layout || type.RowCount() == 3U`. If it is std140 i.e., `use_std430_layout == false`, it is true only when `type.RowCount() == 3U`. In other words, if it is std140, only matrix with 3 elements in a row matters.

Reply:
Right, but even in std140, can we make the row size 2 and the stride 16, or the row size 4 and then then stride is 16 anyway? Basically, can we force these to have the same stride even if we don't have too?

Reply:
When glslang compiles a shader with a matrix whose row size is 2, it always sets the stride 8.
It sets the stride 16 only for matrix with 3 or 4 rows.

If a user correctly writes a vkrunner script, we expect `mat.*` types in SSBO command matches `mat.*` types in shaders.
Thus, in std140, we must set the stride 8 for the row size 2.
We must set the stride 16 for the row size 3 or 4.